### ISSUE TITLE
Add hook to ban users

### ISSUE THREAD

**Issue by nikolay-turpitko** [letsrock-today/authkit#12]
When user makes many attempts to login or many requests with wrong access key or many attempts to change password we should block him. We may mark his account as suspended in user store, provided by app (add methods to block/unblock to API, if necessary).

We should provide mechanisms to detect such attempts, block user and notify application about it.
There should be settings in the lib for this task:
- number of attempts to login, to change password;
- number of attempts to use API with wrong access token;
- callback to notify app (may be callback in user service is already enough? try to implement sample with it);
- whether block user or only notify app (if we just notify app via user service, it may decide for itself, without additional options);

How to unblock user? Should lib provide some schedulers, or leave it to app itself?
Another concern: it may be better to block user at web server level, not on application side to minimize load. So, application may use approach similar to fail2ban, external tool or invoke system calls itself to add iptable or firewall rule to block user's IP and add system timer to unblock it later.

We may add primitives to do this task via calls to systemd, for instance.

**Comment by nikolay-turpitko** [comment]
I decided to postpone this issue. Currently I have no idea how it could be properly implemented in general case. We could only think about some special callback to notify app about failed attempt, but it's not a very big deal, we may add this callbacks when we really need them.

In LR project we want to use authkit as a module for Nginx. Initially on the same host as Nginx with unix socket connection, but potentially we may need to put it onto another node. So, it should invoke some command on different node. In-process communication (callbacks etc) won't work in such case. But module already reports failed attempt to Nginx - in the form of response to the Nginx's request.

I think, (at least for now) it's better to use fail2ban to protect Nginx and app. Especially, because we already use it to protect ssh. Also, this tool is external to the app, so, it could be more easily replaced if required. Fail2ban analyzes nginx or custom logs and best we can do to assist it is to write recognizable errors into log files. I think, we already able to do this. Fail2ban may be customized to implement custom "jails", it seems, that it is possible to implement required scenarios with it.

I think, we may provide some example of fail2ban usage for scenarios above for Nginx module, after we implement module itself.

See https://www.digitalocean.com/community/tutorials/how-to-protect-an-nginx-server-with-fail2ban-on-ubuntu-14-04#configuring-fail2ban-to-monitor-nginx-logs.
